### PR TITLE
Use mmap, mprotect instead of calloc on POSIX system

### PR DESCRIPTION
**Pull request by ksh8281**
* Fix default memory maximium size bug

### CODE REVIEW

File: src/runtime/Memory.cpp [hunk 1]
```diff
@@ -33,29 +38,76 @@ Memory* Memory::createMemory(Store* store, uint64_t initialSizeInByte, uint64_t
 Memory::Memory(uint64_t initialSizeInByte, uint64_t maximumSizeInByte)
     : m_sizeInByte(initialSizeInByte)
     , m_maximumSizeInByte(maximumSizeInByte)
-    , m_buffer(reinterpret_cast<uint8_t*>(calloc(1, initialSizeInByte)))
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/Memory.cpp:36]: Okay!

File: src/runtime/Memory.cpp [hunk 2]
```diff
@@ -33,29 +38,76 @@ Memory* Memory::createMemory(Store* store, uint64_t initialSizeInByte, uint64_t
...
 {
+    RELEASE_ASSERT(initialSizeInByte <= std::numeric_limits<size_t>::max());
+#if defined(WALRUS_USE_MMAP)
+    if (m_maximumSizeInByte) {
+        uint64_t initialReservedSize =
+#if defined(WALRUS_32)
+            (1024) * 1024 * 64;
+#else
+            (1024) * 1024 * 512;
+#endif
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/Memory.cpp:50]: In 32-bit, we can use only 3GB address. 64MB is practical size.
if we use a lots of address, another code in process like malloc will stop to work.

[the remark this answers is not in the document]
Comment (clover2123) [reply src/runtime/Memory.cpp]: I see

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**